src/headers.rs

```diff
@@ -0,0 +1,179 @@
+//! This module provides access to the HTTP MIME headers
+//! of a request or response.
+//!
+//! Not to be confused with the request/response metadata
+//! which is also called "headers" in TrafficServer
+
+use crate::bindings::*;
+use crate::helpers::checked_raw_str;
+use std::marker::PhantomData;
+
+/// An iterator over header field values
+///
+/// Skips any invalid strings
+pub struct HeaderValueIter<'a, 'b, T: 'a> {
+    field: &'b HeaderField<'a, 'b, T>,
+    index: usize,
+    length: usize,
+}
+
+impl<'a, 'b, T> Iterator for HeaderValueIter<'a, 'b, T> {
+    type Item = &'a str;
```

Comment:
See the comment regarding the drop impl. I think, the lifetime `'a` is not the right one here. Instead, it should be `'b` (or `'a : 'b`), because if `HeaderField` is dropped, it will release the backing memory this string points to.